app/services/oneoffs/fix_participant_declaration_user.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Oneoffs
+  class FixParticipantDeclarationUser
+    def call
+      ActiveRecord::Base.transaction do
+        incorrect_participant_declarations.in_batches.each_record do |declaration|
+          if declaration_has_transferred_user?(declaration)
+            declaration.update!(user_id: declaration.participant_profile.user.id)
+          end
+        end
+      end
+    end
+
+    def incorrect_participant_declarations
+      ParticipantDeclaration
+        .joins(participant_profile: :participant_identity)
+        .where("participant_declarations.user_id != participant_identities.user_id")
+    end
+
+    def declaration_has_transferred_user?(declaration)
+      # participant_identities have been moved to another user using Identity::Transfer
+      declaration.user.participant_identities.empty? &&
```

Comment:
Im just curious if we didn't include this, what are the numbers for declarations we need to move across? As thinking maybe we should do that regardless?

Reply:
There are 7208 declarations that have incorrect `user_id`.
81 of return `false` with `declaration_has_transferred_user ?`

Reply:
thanks Mook, I think might be worth checking a few of those incase and if they are ok maybe we change all of them and don't limit to transferred users? Let me know your thoughts :)